src/activities/CategoriesEntity.js

```diff
@@ -165,13 +165,17 @@ export class CategoriesEntity extends Entity {
 		return categoryId !== initialId;
 	}
 
-	async save(category, shouldDeletePreviousCategory = false) {
+	async save(category, shouldDeletePreviousCategory = null) {
 		const hasCategoryIdChanged = category.categoryId && this._hasCategoryIdChanged(category.categoryId);
 
+		if (hasCategoryIdChanged && shouldDeletePreviousCategory !== null) {
+			const { action, fields } = this._generateDeletePreviousAndAssociateWithNewCategory(category, shouldDeletePreviousCategory) || {};
+
+			return action && await performSirenAction(this._token, action, fields);
+		}
+
 		if (hasCategoryIdChanged && category.categoryId !== UNSET_CATEGORY_ID) {
-			const { action, fields } = shouldDeletePreviousCategory
-				? this._generateDeletePreviousAndAssociateWithNewCategory(category, shouldDeletePreviousCategory) || {}
-				: this._generateSelectCategoryAction(category) || {};
+			const { action, fields } = this._generateSelectCategoryAction(category) || {};
```

Comment:
Restored back to old implementation.

Reply:
Do we need both `_generateSelectCategoryAction` and `_generateDeletePreviousAndAssociateWithNewCategory`? Wouldn't one action with the `shouldDeletePreviousCategory` suffice? The value of the field determines whether we delete the previous category or not?

Reply:
`_generateSelectCategoryAction` is needed for Assignments, were you thinking of refactoring this action to use the `shouldDeletePreviousCategory` param?

Reply:
Oh I see, no that makes sense, let's keep it separate then.